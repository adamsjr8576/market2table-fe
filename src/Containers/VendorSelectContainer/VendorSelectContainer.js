import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './VendorSelectContainer.scss';

export const VendorSelectContainer = () => {

  return (
    <div className='vendor-select-container'>
      <section className='vendor-list-section'>
        <h2 className='vendor-select-header'>Vendors</h2>
        <form className='search-vendors-form'>
          <input type='text' className='search-vendors-input' placeholder='Search Vendors...'/>
          <div className='search-vendors-button-container'>
            <button type='button' className='search-vendors-button'>Search</button>
            <button type='button' className='search-vendors-button'>Clear</button>
          </div>
        </form>
        <section className='vendor-cards-section'>
        </section>
      </section>
      <section>
        <button className='create-vendor-button'>Create New Vendor</button>
      </section>
    </div>
  )
}

export default VendorSelectContainer
