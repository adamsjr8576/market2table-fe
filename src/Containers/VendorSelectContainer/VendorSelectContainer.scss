.vendor-select-container {
  height: 85vh;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: space-around;
}

.vendor-list-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  width: 60%;
  background-color: white;
  border: 3px solid darkgrey;
  height: 90%;
  border-radius: 10px;
}

.vendor-select-header {
  font-size: 2em;
  margin: 15px 0;
  width: 30%;
  border-bottom: 4px solid #5E7838;
  text-align: center;
}

.search-vendors-form {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 90%;
}

.search-vendors-input {
  width: 30%;
  border-radius: 20px;
  border: 2px solid black;
  background-color: #F0F0EF;
  padding: 10px;
  outline: 0;
  font-size: 1em;
}

.search-vendors-button-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  height: 75px;
  width: 30%;
}

.search-vendors-button {
  width: 50%;
  padding: 5px;
  font-size: 1em;
  border: 2px solid #3B3D39;
  background-color: #F0F0EF;
  outline: 0;
  border-radius: 10px;
  cursor: pointer;
}

.search-vendors-button:hover {
  background-color: #C4D1B2;
  border: 2px solid #5E7838;
}

.vendor-cards-section {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  height: 75%;
}
