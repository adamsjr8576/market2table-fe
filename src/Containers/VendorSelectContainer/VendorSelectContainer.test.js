import React from 'react';
import { shallow } from 'enzyme';
import VendorSelectContainer from './VendorSelectContainer';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
    vendors: [{name:'vendorA', description:'selling', products:'apples'},{name:'vendorB', description:'sellin', products:'oranges'}],
  }
jest.mock("react-redux", () => ({
    useSelector: () => mockState,
    useDispatch: () => jest.fn()
}));

describe('VendorSelectContainer', () => {
    let mockAddVendors = jest.fn()
    it('should match the ZipCodeForm snapshot', () => {
        let wrapper = shallow(<VendorSelectContainer
            vendors={mockState.vendors}
            addVendors={mockAddVendors}
          />);
        expect(wrapper).toMatchSnapshot();
    })

});