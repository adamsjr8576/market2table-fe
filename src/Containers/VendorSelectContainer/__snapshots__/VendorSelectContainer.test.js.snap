// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VendorSelectContainer should match the ZipCodeForm snapshot 1`] = `
<div
  className="vendor-select-container"
>
  <section
    className="vendor-list-section"
  >
    <h2
      className="vendor-select-header"
    >
      Vendors
    </h2>
    <form
      className="search-vendors-form"
    >
      <input
        className="search-vendors-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search Vendors..."
        type="text"
        value=""
      />
      <div
        className="search-vendors-button-container"
      >
        <button
          className="search-vendors-button"
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
        <button
          className="search-vendors-button"
          onClick={[Function]}
          type="button"
        >
          Clear
        </button>
      </div>
    </form>
    <div
      className="vendor-search-error-container"
    >
      <p
        className="vendor-search-error"
        hidden={true}
      >
        No matches for vendor name: 
      </p>
    </div>
    <section
      className="vendor-cards-section"
    />
  </section>
  <Link
    className="link-create-vendor-form"
    to="/vendor/account/form"
  >
    <button
      className="create-vendor-button"
    >
      <p
        className="vendor-button-p"
      >
        Create New Vendor
      </p>
      <img
        alt="plus sign icon"
        className="plus-icon"
        src="plus.png"
      />
    </button>
  </Link>
</div>
`;
